make/autoconf/libraries.m4

```diff
@@ -149,6 +149,13 @@ AC_DEFUN_ONCE([LIB_SETUP_LIBRARIES],
     fi
   fi
 
+  # Programs which use C11 or C++11 atomics, like #include <atomic>,
```

Comment:
I think the old code comment here is a bit too general. It does not mean we introduce any use of C++ atomics here.
The fact is that RISC-V only has word-sized atomics, it requries libatomic where other common architectures do not [1].
So atomic support would require explicit linking against -latomic on RISC-V. Otherwise we got build errors like:

```
/usr/bin/ld: /home/yangfei/jdk-riscv-port/build/linux-riscv64-server-slowdebug/hotspot/variant-server/libjvm/objs/constantPool.o: in function `signed char Atomic::PlatformCmpxchg<1ul>::operator()<signed char>(signed char volatile*, signed char, signed char, atomic_memory_order) const':
/home/yangfei/jdk-riscv-port/src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp:75: undefined reference to `__atomic_compare_exchange_1'
/usr/bin/ld: /home/yangfei/jdk-riscv-port/build/linux-riscv64-server-slowdebug/hotspot/variant-server/libjvm/objs/freeListAllocator.o: in function `bool Atomic::PlatformCmpxchg<1ul>::operator()<bool>(bool volatile*, bool, bool, atomic_memory_order) const':
/home/yangfei/jdk-riscv-port/src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp:75: undefined reference to `__atomic_compare_exchange_1'
/usr/bin/ld: /home/yangfei/jdk-riscv-port/build/linux-riscv64-server-slowdebug/hotspot/variant-server/libjvm/objs/javaClasses.o: in function `unsigned char Atomic::PlatformCmpxchg<1ul>::operator()<unsigned char>(unsigned char volatile*, unsigned char, unsigned char, atomic_memory_order) const':
/home/yangfei/jdk-riscv-port/src/hotspot/os_cpu/linux_riscv/atomic_linux_riscv.hpp:75: undefined reference to `__atomic_compare_exchange_1'
```

I have updated the code comment in my last commit. Hope that addresses your concerns :-)

P.S. : I realized that -latomic is not needed for BASIC_JDKLIB_LIBS here, so only add this link option to BASIC_JVM_LIBS.

[1] https://github.com/riscv-collab/riscv-gcc/issues/12

Reply:
New comment looks good - thanks for clarifying.